packages/input-text/src/TextInput.js

```diff
@@ -0,0 +1,165 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import Tooltip from "@crave/farmblocks-tooltip";
+import Image, { badgeSizes } from "@crave/farmblocks-image";
+
+import errorIconSrc from "./constants/errorIcon";
+import StyledLabel from "./styledComponents/TextInput";
+
+class TextInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.value,
+      invalid: this.props.invalid,
+      showTooltip: false
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.onInvalid = this.onInvalid.bind(this);
+    this.onMouseOver = this.onMouseOver.bind(this);
+    this.onMouseLeave = this.onMouseLeave.bind(this);
+  }
+
+  render() {
+    const {
+      label,
+      value,
+      onChange,
+      invalid,
+      validationErrors,
+      onInvalid,
+      tooltipText,
+      errorIconSrc,
+      onMouseOver,
+      onMouseLeave,
+      ...otherProps
+    } = this.props;
+    const containerProps = {
+      filled: this.state.value.length > 0,
+      invalid: this.state.invalid,
+      disabled: otherProps.disabled
+    };
+    const inputProps = {
+      value: this.state.value,
+      onChange: this.onChange,
+      onInvalid: this.onInvalid,
+      onMouseOver: this.onMouseOver,
+      onMouseLeave: this.onMouseLeave,
+      ...otherProps
+    };
+    return (
+      <StyledLabel {...containerProps}>
+        {this._renderInput(inputProps)}
+        {this._renderLabel(label)}
+        {this._renderTooltip(this.state.showTooltip, tooltipText)}
+        {this._renderFailedMessages(
+          this.state.invalid,
+          validationErrors,
+          errorIconSrc
+        )}
+      </StyledLabel>
+    );
+  }
+
+  _renderLabel(label) {
+    return label && <div className="label">{label}</div>;
+  }
+
+  _renderInput(inputProps) {
+    return <input type="text" {...inputProps} />;
+  }
+
+  _renderTooltip(visible, text) {
+    return (
+      visible && (
+        <div className="tooltip">
+          <Tooltip text={text} />
+        </div>
+      )
+    );
+  }
+
+  _renderFailedMessages(invalid, errors, iconSrc) {
+    return (
+      invalid && (
+        <div className="messages">
+          {errors.map((text, index) => (
+            <div className="message" key={`err-${index}`}>
+              <Image
+                className="icon"
+                src={iconSrc}
+                badge
+                size={badgeSizes.SMALL}
+              />
+              {text}
```

Comment:
Image and text are misaligned...
It should be vertically center aligned.

Also use `@crave/farmblocks-text`component to render all texts. This way we can follow a pattern and centralize the text styling.

![image](https://user-images.githubusercontent.com/17936244/33731339-eb6a17c8-db69-11e7-97c5-6ddbddafb896.png)

Reply:
I have fixed this alignment by setting the text line height to be the same of the image

Reply:
I have added farmblocks-text to the dependencies and have used for label and for the error message, but I still had to add some rules on top of that, like the letter-spacing from your other request